clients/src/test/java/org/apache/kafka/common/network/SelectorTest.java

```diff
@@ -837,8 +843,9 @@ private void sendAndReceive(String node, String requestPrefix, int startIndex, i
                 responses++;
             }
 
-            for (int i = 0; i < selector.completedSends().size() && requests < endIndex; i++, requests++) {
```

Comment:
I'm not sure what the original intent of this code was.  It seems like if there is ever a poll() that doesn't return any completed sends, we could get stuck here since we don't send a new request in that case.  poll() can return for many reasons, not all of which involve completed sends, so this seems wrong to me.  The new version always succeeds for me.